exercises/clock/.meta/generator/clock_case.rb

```diff
@@ -9,27 +9,49 @@ def name
   end
 
   def workload
-    property == 'equal' ? compare_clocks : simple_test
+    case property
+    when 'create'
+      simple_test
+    when 'add'
+      add_to_clock
+    when 'subtract'
+      subtract_from_clock
+    when 'equal'
+      compare_clocks
+    else
+      raise "Encountered unknown property in canonical-data.json"
+    end
   end
 
   private
 
   def compare_clocks
-    "clock1 = Clock.at(#{clock1['hour']}, #{clock1['minute']})
-    clock2 = Clock.at(#{clock2['hour']}, #{clock2['minute']})
+    first_hour = input['clock1']['hour']
+    first_minute = input['clock1']['minute']
+    second_hour = input['clock2']['hour']
+    second_minute = input['clock2']['minute']
+
+    "clock1 = Clock.new(hour: #{first_hour}, minute: #{first_minute})
+    clock2 = Clock.new(hour: #{second_hour}, minute: #{second_minute})
     #{assert} clock1 == clock2"
   end
 
   def simple_test
     [
       "assert_equal #{expected.inspect}, ",
-      "#{'(' if add_to_clock}Clock.at(#{hour}, ",
-      "#{minute})#{add_to_clock}#{')' if add_to_clock}.to_s"
+      "Clock.new(#{"hour: #{input['hour']}" unless input['hour'] == 0}",
+      "#{", " unless input['hour'] == 0 || input['minute'] == 0}",
+      "#{"minute: #{input['minute']}" unless input['minute'] == 0}).to_s"
     ].join
   end
 
   def add_to_clock
-    " + #{add}" if respond_to?(:add)
+    "clock1 = Clock.new(hour: #{input['hour']}, minute: #{input['minute']})
+    assert_equal #{expected.inspect}, (clock1 + Clock.new(minute: #{input['value']})).to_s"
```

Comment:
Here we're relying on the fact that the indentation of this file is the same as the indentation of the final test file, but this is just a coincidence.
You probably should use the [indent_lines helper method](https://github.com/exercism/ruby/blob/master/lib/generator/exercise_case/case_helpers.rb#L14)

Reply:
Those that return multiple lines of output anyway.

Reply:
Got `indent_lines` on three of them, and used `indent_test` on the one line test for good measure.